src/rest/server_service/path_predicate.rs

```diff
@@ -0,0 +1,79 @@
+use actix_web::pred::Predicate;
+use actix_web::Request;
+use regex::{self, Regex};
+
+pub struct PathPredicate {
+    regex: Regex,
+}
+
+impl PathPredicate {
+    // The pattern should be a full URI, with or without slash on the beginning.
+    // Wildcards in form of `{some_name}` are allowed, they accept everything except `/`.
+    // The predicate will accept only requests with URIs fully matching the pattern.
+    pub fn for_pattern(pattern: &str) -> Self {
+        let pattern = regex::escape(&pattern);
+        let segment_regex = Regex::new(r"\\\{[^}]*\}").unwrap();
+        let pattern = segment_regex.replace_all(&pattern, r"[^/]*");
```

Comment:
Ah, I get it now. So all this complexity and regex-tweaking-regex is just to have some `{foo}` constructs in your pattern that you then replace with simple matches? Why have them and the pattern syntax in the first place?

Reply:
I think this question was answered in other thread: https://github.com/input-output-hk/jormungandr/pull/168#discussion_r263232925